dmscripts/upload_dos_opportunities_email_list.py

```diff
@@ -31,14 +27,14 @@ def main(data_api_client, mailchimp_client, lot_data, logger):
         "{} emails have been found for lot {}".format(len(emails), lot_data["lot_slug"])
     )
 
-    existing_mailchimp_emails = mailchimp_client.get_email_addresses_from_list(lot_data["list_id"])
+    existing_mailchimp_emails = list(mailchimp_client.get_email_addresses_from_list(lot_data["list_id"]))
     logger.info(
         "{} existing emails found on list {}".format(len(existing_mailchimp_emails), lot_data["list_id"])
     )
 
     # lowercase required because mailchimp may capatalise emails addresses returned from the mailchimp API based on it's
     # on how it stores both the lowercase hash but also the original (potentially capatilised) email address
-    new_emails = list(set(lowercase_list(emails)) - set(lowercase_list(existing_mailchimp_emails)))
+    new_emails = set(map(lambda x: x.lower(), emails)) - set(map(lambda x: x.lower(), existing_mailchimp_emails))
```

Comment:
And if you used `set.difference` here, you wouldn't need to construct the second set *at all*. (Although again this probably won't make any difference, if we're optimising...)

Also the first set could be a frozenset :)